core/src/test/java/io/grpc/ContextTest.java

```diff
@@ -555,99 +554,108 @@ public void typicalCancellableTryCatchFinallyHandling() throws Exception {
       base.detachAndCancel(previous, null);
     }
     assertTrue(base.isCancelled());
-    assertNotNull(base.cause());
+    assertNotNull(base.cancellationCause());
   }
 
   @Test
-  public void testAttachAsCloseable() throws Exception {
-    Context current = Context.current();
-    Context.CancellableContext base = Context.current().withValue(FOOD, "fish").withCancellation();
-
-    Context previous = base.attach();
-    Context baseAttached = Context.current();
-    // Should not be able to get back the CancellableContext
-    assertNotSame(baseAttached, base);
-    base.detach(previous);
-
-    Closeable c = base.attachAsCloseable();
-    assertEquals("fish", FOOD.get());
-    assertFalse(base.isCancelled());
-    assertNull(base.cause());
-
-    c.close();
-    assertSame(current, Context.current());
-    assertTrue(base.isCancelled());
-    assertNull(base.cause());
+  public void rootHasNoDeadline() {
+    assertNull(Context.ROOT.getDeadline());
   }
 
-  /*
-  public void testTryWithResource() throws Exception {
-    Context.CancellableContext base = Context.current().withCancellation();
+  public void contextWithDeadlineHasDeadline() {
+    Context.CancellableContext cancellableContext =
+        Context.ROOT.withDeadlineAfter(1, TimeUnit.SECONDS, scheduler);
+    assertNotNull(cancellableContext.getDeadline());
+  }
 
-    try (Closeable c = base.attachAsCloseable()) {
-      // Do something
-      throw new IllegalStateException("Argh");
-    } catch (IllegalStateException ise) {
-      // Don't capture exception
+  @Test
+  public void earlierParentDeadlineTakesPrecedenceOverLaterChildDeadline() throws Exception {
+    final Deadline sooner = Deadline.after(2, TimeUnit.SECONDS);
```

Comment:
Wouldn't a deadline after 0 s, 1 ms, or 100 ms be equally effective without taking as long?